### PR TITLE
Extrication filter

### LINKED ISSUE #24: Implement extrication trajectory filter

### PR DESCRIPTION
**Pull request by Nandite**
@VictorLamoine, @KevinBollore Added extrication filter function. Tests have been made on planes, cylinder and DAC meshes, everything work fine ! 
Fixes #24 !

### CODE REVIEW

File: bezier_library/src/bezier_grinding_surfacing.cpp [hunk 1]
```diff
@@ -183,7 +183,17 @@ std::string BezierGrindingSurfacing::generateTrajectory(EigenSTL::vector_Affine3
     EigenSTL::vector_Affine3d traj;
     if (!generateRobotPosesAlongStripper(*it, traj))
       return "Could not generate robot poses for extrication trajectory";
-
+    Eigen::Affine3d first_pose(traj.front());
+    Eigen::Affine3d last_pose(traj.back());
+    Eigen::Vector3d first_point(first_pose.translation());
+    Eigen::Vector3d last_point(last_pose.translation());
+    Eigen::Vector3d first_point_normal(first_pose.linear().col(2));
+    Eigen::Vector3d last_point_normal(last_pose.linear().col(2));
```
Comment (VictorLamoine) [inline bezier_library/src/bezier_grinding_surfacing.cpp]: I would delete first_pose and last_pose and directly use traj.front() etc. in the point/normal in the constructors

File: bezier_library/src/bezier_grinding_surfacing.cpp [hunk 2]
```diff
@@ -598,23 +605,37 @@ bool BezierGrindingSurfacing::filterExtricationTrajectory(const vtkSmartPointer<
       start_of_intermediate_part = it;
       break; // Break the loop process
     }
+    if (scalar_res > upper_bound)
+    {
+      // This loop is executed whenever the result computed by the dot product
+      // has overshoot the bounds limit
```
Comment (VictorLamoine) [inline bezier_library/src/bezier_grinding_surfacing.cpp]: Please explain the meaning of the scalar result being superior to the upper bound. (= the point is too far).

Please explain what we do in this case (we delete/ignore this point)

**Comment by VictorLamoine** [comment]
I'm not sure to understand how this works, I'll take a deeper look tomorrow

**Comment by KevinBollore** [comment]
It's in construction, doesn't work yet ( that's why I ask to Libasse to put a label :construction: Work in progress :construction: ).

**Comment by VictorLamoine** [comment]
👌

**Pull request merged**